lib/spack/spack/util/environment.py

```diff
@@ -744,11 +744,11 @@ def from_sourcing_file(
             ]
         )
 
+        # Implements a no op during forwards operation on respective os
+        null_fd = "cd %CD%" if sys.platform == "win32" else os.devnull
```

Comment:
Can we change this to `echo > NUL`. It achieves the same thing as `cd %CD%` with the added benefit of being a bit more semantically in line with the intent of the operation here.